src/main/java/com/microsoft/sqlserver/jdbc/SQLServerVSMEnclaveProvider.java

```diff
@@ -275,9 +276,10 @@ void validateCert(byte[] b) throws SQLServerException {
                         .generateCertificates(new ByteArrayInputStream(b));
                 for (X509Certificate cert : certs) {
                     try {
+                        cert.checkValidity();
                         healthCert.verify(cert.getPublicKey());
                         return;
-                    } catch (SignatureException e) {
+                    } catch (SignatureException | CertificateExpiredException e) {
                         // Doesn't match, but continue looping through the rest of the certificates
```

Comment:
should we log a warning?

Reply:
I don't think it's necessary, up to the user to make sure their certificates haven't expired.

Reply:
I think would be helpful if someone complains that they do have the certificate why isn't it working. But, I don't feel strongly about it so ok.